docs/src/Submakefile

```diff
@@ -478,36 +478,35 @@ SE := \" style=\"-moz-column-width: 25ex; \
 #     * the manpage section title
 #
 ADD_HTML_MANPAGES = \
-	@echo "Adding manpages: man$(strip $(1)): $(strip $(2))"; \
-	echo "$(SA)$(strip $(1))$(SB)$(strip $(1))$(SC)$(strip $(2))</a></p>" >> objects/index.incl; \
-	echo "$(SD)$(strip $(1))$(SE)" >> objects/index.incl; \
+	echo "Adding manpages: man$(strip $(2)): $(strip $(3))"; \
+	echo "$(SA)$(strip $(2))$(SB)$(strip $(2))$(SC)$(strip $(3))</a></p>" >> objects/index.incl; \
+	echo "$(SD)$(strip $(2))$(SE)" >> objects/index.incl; \
 	echo "<ul>" >> objects/index.incl; \
-	for HTML_FILE in $(filter $(DOC_DIR)/html/man/man$(strip $(1))/%.html, $(MAN_HTML_TARGETS)); do \
+	for HTML_FILE in $(filter $(DOC_DIR)/html/man/man$(strip $(1))/%.$(strip $(2)).html, $(MAN_HTML_TARGETS)); do \
 		BASENAME=$$(basename $$HTML_FILE .html); \
 		echo "<li><a href=\"$${HTML_FILE\#$(DOC_DIR)/html/}\">$${BASENAME%.*}</a></li>"; \
 	done >> objects/index.incl; \
-	echo "</ul></div>" >> objects/index.incl
+	echo "</ul></div>" >> objects/index.incl; \
 
 
 objects/index.incl: $(GENERATED_MANPAGES) objects/var-MAN_HTML_TARGETS $(DOC_SRCDIR)/Submakefile
 	rm -f $@
-	$(call ADD_HTML_MANPAGES, 1, Commands and userspace components)
-	$(call ADD_HTML_MANPAGES, 9, Realtime components and kernel modules)
-	$(call ADD_HTML_MANPAGES, 3hal, API calls: HAL)
-	$(call ADD_HTML_MANPAGES, 3rtapi, API calls: RTAPI)
-	$(call ADD_HTML_MANPAGES, 3hm2,  API calls: Hostmot2)
-	$(call ADD_HTML_MANPAGES, 3, API calls: General)
-
-	# now make sure all manpages made it into the html index
+	$(call ADD_HTML_MANPAGES, 1, 1, Commands and userspace components) \
+	$(call ADD_HTML_MANPAGES, 9, 9, Realtime components and kernel modules) \
+	$(call ADD_HTML_MANPAGES, 3, 3hal, API calls: HAL) \
+	$(call ADD_HTML_MANPAGES, 3, 3rtapi, API calls: RTAPI) \
+	$(call ADD_HTML_MANPAGES, 3, 3hm2,  API calls: Hostmot2) \
+	$(call ADD_HTML_MANPAGES, 3, 3, API calls: General) \
+	# now make sure all manpages made it into the html index \
 	FAIL=0; \
-	for F in $$(find $(DOC_DIR)/man/? -type f); do \
+	for F in $$(find $(DOC_DIR)/man/man* -type f); do \
 		B=$$(basename $$F); \
 		if ! grep -q $$B $@; then \
 			echo stray manpage not added to index: $$B; \
 			FAIL=1; \
 		fi; \
 	done; \
-	if [ $$FAIL -ne 0 ]; then false; fi
+	if [ $$FAIL -ne 0 ]; then exit 1; fi
```

Comment:
This change look good to me, and I suspect it is working as intended.  But I wonder if the for loop perhaps would be more specific if it used $(DOC_DIR)/man/man? instead of $(DOC_DIR)/man/man*, and just did 'exit 1' inside the for loop instead of the FAIL=1 stuff.  Why do you want to process all man page paths before exiting?

Reply:
@petterreinholdtsen your example seems to be accidentally formatted like Latex or math or so. Can you fix this to make your comment more clear?

Reply:
Sorry, did not notice this.  Fixed.

Reply:
@jepler what do you think? Would be nice to get this merged.

Reply:
Suppose you mess up 37 things. Without FAIL= you have to make 37 times and fix one thing each time. However, please do whatever you care to do.

Reply:
Anyway I didn't introduce FAIL= I just corrected its use so that it causes the overall make process to fail.